lib/rpc/ProdConfig.ts

```diff
@@ -30,11 +30,22 @@ export const ProdConfigJoi = Joi.array().items(
   })
 )
 
-export function getRpcGatewayEnabledChainIds(): ChainId[] {
-  const validation = ProdConfigJoi.validate(PROD_CONFIG)
+// Return a map of chain id and its provider names
+export function getRpcGatewayEnabledChains(config?: object): Map<ChainId, string[]> {
+  const prodConfigInput = config !== undefined ? config : PROD_CONFIG
+  const validation = ProdConfigJoi.validate(prodConfigInput)
   if (validation.error) {
-    throw new Error(`ProdConfig failed data validation: Value: ${PROD_CONFIG}, Error: ${validation.error.message}`)
+    throw new Error(`ProdConfig failed data validation: Value: ${prodConfigInput}, Error: ${validation.error.message}`)
   }
   const prodConfig: ProdConfig = validation.value as ProdConfig
-  return prodConfig.map((chainConfig) => chainConfig.chainId)
+  let result = new Map<ChainId, string[]>()
+  for (const chainConfig of prodConfig) {
+    if (chainConfig.providerUrls) {
+      result.set(
+        chainConfig.chainId,
+        chainConfig.providerUrls.map((url: string) => url.substring(0, url.indexOf('_')))
```

Comment:
This substring is fairly unreadable. Can you add some comments?

Reply:
Yes, comments added.